cpu/stm32/periph/adc_f0_g0.c

```diff
@@ -75,6 +75,19 @@ int adc_init(adc_t line)
     }
     /* reset configuration */
     ADC1->CFGR2 = 0;
+    ADC1->CR |= ADC_CR_ADVREGEN;
+    ADC1->CR &= ~ADC_CR_ADCAL;
+    // calibrate ADC, per RM0454 section 14.3.3
+    // 1. ensure ADEN=0, ADVREGEN=1, DMAEN=0
+    if (!(ADC1->CR & ADC_CR_ADEN) && (ADC1->CR & ADC_CR_ADVREGEN) && !(ADC1->CFGR1 & ADC_CFGR1_DMAEN))
+    {
+        // 2. Set ADCAL=1
+        ADC1->CR |= ADC_CR_ADCAL;
+        // 3. Wait for ADCAL=0 (or EOCAL=1)
+        while ((ADC1->ISR & ADC_ISR_EOCAL))
+        {
+        }
```

Comment:
```suggestion
        while ((ADC1->ISR & ADC_ISR_EOCAL)) {}
```

Reply:
I haven't timed it, but it exited immediately in the brief testing I did.